devops/lib/component.py

```diff
@@ -84,20 +84,29 @@ def validate(self, ctx=None):
             if result.returncode > 0:
                 raise ValidationError(f"Validation failed for {path}")
 
-    def build(self, ctx: Context, dry_run=False):
+    def build(self, ctx: Context, dry_run=False, docker_args=None):
         label(logger.info, f"Building {self.path}")
         dockerfile = self.path / "Dockerfile"
 
         if not dockerfile.exists():
             logger.info(f"No Dockerfile for {self.name} component")
             return
 
+        if isinstance(docker_args, list):
+            # Insert --build-arg before each item in docker_args.
+            docker_args = [["--build-arg", x] if x else "" for x in docker_args]
+            # Flatten list
+            docker_args = [y for x in docker_args for y in x]
```

Comment:
Should this not filter the `""`s out?

Reply:
Please use better names than `x` and `y`

Reply:
Answered in `Why is this if x required?` comment.